indexer/feature.cpp

```diff
@@ -89,15 +94,16 @@ int GetScaleIndex(SharedLoadInfo const & loadInfo, int scale,
     int const lastScale = loadInfo.GetLastScale();
     if (scale > lastScale)
       scale = lastScale;
-    // If there is no geometry for the requested scale
-    // fallback to the next more detailed one.
-    while (ind < count && (scale > loadInfo.GetScale(ind) || offsets[ind] == kInvalidOffset))
+
+    // If there is no geometry for the requested scale (kHasGeoOffsetFlag) fallback to the next more detailed one.
+    while (ind < count && (scale > loadInfo.GetScale(ind) || offsets[ind] == kGeomOffsetFallback))
       ++ind;
+
     // Some WorldCoasts features have idx == 0 geometry only and its possible
     // other features to be visible on e.g. idx == 1 only,
     // but then they shouldn't be attempted to be drawn using other geom scales.
     ASSERT_LESS(ind, count, ("No suitable geometry scale range in the map file."));
-    return (ind < count ? ind : -1);
+    return (offsets[ind] != kInvalidOffset ? ind : -1);
```

Comment:
Is fallback offset ok here?

Reply:
We won't get here with kGeomOffsetFallback